src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/LicenseScanTests.cs

```diff
@@ -0,0 +1,243 @@
+        "cecill-c", // https://cecill.info/licences/Licence_CeCILL-C_V1-en.txt
+        "epl-1.0", // https://opensource.org/license/epl-1-0/
+        "generic-cla", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/generic-cla.LICENSE
+        "gpl-1.0-plus", // https://opensource.org/license/gpl-1-0/
+        "gpl-2.0", // https://opensource.org/license/gpl-2-0/
+        "gpl-2.0-plus WITH autoconf-simple-exception-2.0", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/rules/gpl-2.0-plus_with_autoconf-simple-exception-2.0_8.RULE
+        "gpl-2.0 WITH gcc-linking-exception-2.0", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/rules/gpl-2.0_with_gcc-linking-exception-2.0_6.RULE
+        "isc", // https://opensource.org/license/isc-license-txt/
+        "json", // https://www.json.org/license.html
+        "lgpl-2.0-plus", // https://opensource.org/license/lgpl-2-0/
+        "lgpl-2.1", // https://opensource.org/license/lgpl-2-1/
+        "lgpl-2.1-plus", // https://opensource.org/license/lgpl-2-1/
+        "mit", // https://opensource.org/license/mit/
+        "mit-addition", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/mit-addition.LICENSE
+        "ms-pl", // https://opensource.org/license/ms-pl-html/
+        "ms-rl", // https://opensource.org/license/ms-rl-html/
+        "newton-king-cla", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/newton-king-cla.LICENSE
+        "ngpl", // https://opensource.org/license/nethack-php/
+        "ofl-1.1", // https://opensource.org/license/ofl-1-1/
+        "osf-1990", // https://fedoraproject.org/wiki/Licensing:MIT?rd=Licensing/MIT#HP_Variant
+        "public-domain", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/public-domain.LICENSE
+        "public-domain-disclaimer", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/public-domain-disclaimer.LICENSE
+        "rpl-1.5", // https://opensource.org/license/rpl-1-5/
+        "unicode", // https://opensource.org/license/unicode-inc-license-agreement-data-files-and-software/
+        "uoi-ncsa", // https://opensource.org/license/uoi-ncsa-php/
+        "warranty-disclaimer", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/warranty-disclaimer.LICENSE
+        "zlib" // https://opensource.org/license/zlib/
+    };
+
+    private static readonly string[] s_ignoredFilePatterns = new string[]
+    {
+        "*.bin",
+        "*.bmp",
+        "*.bson",
+        "*.db",
+        "*.dic",
+        "*.docx",
+        "*.eot",
+        "*.gif",
+        "*.ico",
+        "*.jpg",
+        "*.il",
+        "*.ildump",
+        "*.lss",
+        "*.nlp",
+        "*.otf",
+        "*.pdf",
+        "*.pfx",
+        "*.png",
+        "*.rtf",
```

Comment:
I understand the reason for trying to exclude binary files, but some rtf (and docx?) files include licenses that we need to deal with. For example, https://github.com/dotnet/roslyn/blob/main/src/Setup/Roslyn.ThirdPartyNotices/ThirdPartyNotices.rtf includes several licenses. This roslyn third party licenses all seem to be open source, but if they were not, it would be good to flag that and then try and address it.

Reply:
Fixed in https://github.com/dotnet/installer/pull/17442/commits/8a464f2e4ba919c5314591993f3d83a605966038